### PR TITLE
feat(application, attendee_application): add GET and POST for application and user models

### LINKED ISSUE #2: feat(): add GET and POST views for user and application models

### LINKED ISSUE #4: feat(): implement file upload

### PR DESCRIPTION
**Pull request by stonewallstan**
Closes #2
Closes #4

### CODE REVIEW

**Comment by mianuddin** [comment]
1. What is the difference between `application` and `attendee-application`?
2. What are the routes that I should be able to test?

**Comment by stonewallstan** [comment]
The attendee-application is the basic settings for the django app. Application is where the true REST framework and the rest of the django app lies. Most of the code that will be modified for the for the project will be in application.

File: application/serializers.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+       model = Application
+       fields = ('url', 'id','first_name', 'last_name', 'email', 
+       'phone_number','is_eighteen', 'school', 'graduation_date', 'major',
+        'city', 'dietary_restrictions', 'allergies', 'github', 'linkedin',
+        'personal_website', 'resume', 'short_answer', 'gender', 'ethnicity',
+        'anything_else', 'agreed')
+class UserSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = User
+        fields= ('url', 'id', 'username', 'password')
```
Comment (mianuddin) [inline application/serializers.py]: Missing space after fields

**Comment by mianuddin** [comment]
What are you currently using for linting?

**Comment by stonewallstan** [comment]
I'm currently using pylint, but I guess it didn't catch that. I'll make sure that it is active in my code.

**Comment by mianuddin** [comment]
screenshot_2018-08-14 operationalerror at application

- Looks like the application route doesn't work
- Could you describe how I can use the user api?

**Comment by stonewallstan** [comment]
Could you elaborate on what you mean by using the user api (like the route you need)?

**Comment by stonewallstan** [comment]
Also, pylint is working, but I think I am probably going to switch the linter since it didn't catch the styling issue.

**Comment by mianuddin** [comment]
How do I create a new user?
 * Where do I `POST`?
 * What information is required?

How do I access information about a specific user? How can I get a list of all users?

**Comment by mianuddin** [comment]
As the user model isn't complete, do you mind removing it for this PR? It'll probably be implemented differently due to auth0 anyway. I tried the application and it looks great. I can `POST` from the browser and can `GET` my entry.

Do you know why this is failing? I copied the body of the message directly from the webpage.
screen shot 2018-08-14 at 19 48 53

**Comment by stonewallstan** [comment]
I don't know why at the moment, but I'm looking into it.

**Comment by stonewallstan** [comment]
I think it was because I didn't have a JSON Parser in my parser classes. It should work now.

**Comment by mianuddin** [comment]
I figured it out, currently it's only set to accept form data.
screen shot 2018-08-14 at 20 16 25

**Comment by mianuddin** [comment]
Works great!
screen shot 2018-08-14 at 20 26 25

**Comment by mianuddin** [comment]
I haven't tried it out with the JSONParser, I'll have to pull again.

**Comment by stonewallstan** [comment]
I think you might have just tried it with the JSONParser included, but not sure. It should still work though.

**Comment by mianuddin** [comment]
Is there any update on JSONParser (aside from file upload concerns)?

**Comment by stonewallstan** [comment]
Everything except the file upload is working with the JSONParser as far as I am aware. I'll change the variable to test the rest of the parser to make sure the parser is working properly.

**Comment by stonewallstan** [comment]
image
Everything else works properly

**Comment by mianuddin** [comment]
Heads up @ohnoitsjmo

**Pull request merged**